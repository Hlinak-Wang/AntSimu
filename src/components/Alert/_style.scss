.alert {
  position: relative;
  width: 100%;
  border-radius: $border-radius;
  font-weight: $font-weight-light;
  padding: 8px 15px;
  margin-bottom: 16px;
  span {
    display: block;
  }
  &.alert-closable {
    padding-right: 30px;
  }
}

.alert-success {
  background: lighten($green,25%);
  border: 0.5px solid darken($green,15%);
}

.alert-error {
  background: lighten($red,25%);
  border: 0.5px solid darken($red,15%);
}

.alert-warning {
  background: lighten($yellow,25%);
  border: 0.5px solid darken($yellow,15%);
}

.alert-info {
  background: lighten($primary,25%);
  border: 0.5px solid darken($primary,15%);
}

.alert-close-button {
  background: transparent;
  border: none;
  overflow: hidden;
  position: absolute;
  top: 8px;
  right: 16px;
  font-size: 12px;
  line-height: 22px;
  cursor: pointer;
  &:focus {
    outline: none;
  }
}


.alertPop-enter {
  opacity: 0;
}
.alertPop-enter-active {
  opacity: 1;
  transition: opacity 500ms;
}

.alertPop-enter-done {
  opacity: 1;
}

.alertPop-exit {
  opacity: 1;
}
.alertPop-exit-active {
  opacity: 0;
  transition: opacity 500ms;
}

.alertPop-exit-done {
  opacity: 0;
}