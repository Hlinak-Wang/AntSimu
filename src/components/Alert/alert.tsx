import React, { useState, useEffect } from 'react'
import { CSSTransition } from 'react-transition-group';
import classnames from 'classnames';

export type AlertType = 'success' | 'info' | 'error' | 'warning';

interface baseAlertProps {
  className?:string;
  style?:React.CSSProperties;
  type: AlertType;
  message: React.ReactNode;
  description?: React.ReactNode;
  closable?: boolean;
  closeText?: React.ReactNode;
  onClose?:React.MouseEventHandler<HTMLButtonElement>;
  afterClose?:() => void;
  onClick?:React.MouseEventHandler<HTMLDivElement>;
  onMouseEnter?:React.MouseEventHandler<HTMLDivElement>;
  onMouseLeave?:React.MouseEventHandler<HTMLDivElement>;
}


const Alert: React.FC<baseAlertProps> = (props) => {
  const { 
    className,
    style,
    type, 
    message, 
    description, 
    closable, 
    closeText,
    onClose,
    afterClose,
    onClick,
    onMouseEnter,
    onMouseLeave
  } = props;
  const [showState, setStatus] = useState(true);
  const handleOnClose:React.MouseEventHandler<HTMLButtonElement> = e => {
    e.preventDefault();
    setStatus(!showState)
    if (onClose) {
      onClose(e)
    }
  }

  let classes = classnames('alert', className, {
    [`alert-${type}`]: true,
    'alert-closable': closable
  })

  if (closable) {
    return (
      <CSSTransition
        in={showState}
        timeout={300}
        classNames="pull-out-top"
        unmountOnExit
        onExited={afterClose}
      >
        <div 
          className={classes}
          style={style}
          onClick={onClick}
          onMouseEnter={onMouseEnter}
          onMouseLeave={onMouseLeave}
        >
          <span className="alert-message">
            {message}
          </span>
          <span className="alert-description">
            {description}
          </span>
            <button className="alert-close-button" onClick={handleOnClose}>
                {closeText}
            </button>
          }
        </div>
      </CSSTransition>
    )
  } else {
    return (
      <div 
        className={classes}
        style={style}
        onClick={onClick}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
        <span className="alert-message">
          {message}
        </span>
        <span className="alert-description">
          {description}
        </span>
      </div>
    )
  }
  
}

Alert.defaultProps = {
  closable: false,
  closeText: "X"
}
export default Alert;