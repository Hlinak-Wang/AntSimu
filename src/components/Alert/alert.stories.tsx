import React from 'react';
import Alert from './alert';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

storiesOf('Alert 警告', module)
  .add('基本样式', () => (
    <>
      <Alert type="success" message="success alert"/>
      <Alert type="info" message="info alert"/>
      <Alert type="error" message="error alert"/>
      <Alert type="warning" message="warning alert"/>
    </>
  ))
  .add('可关闭的警告', () => (
    <>
      <Alert type="success" message="success alert" closable/>
      <Alert type="info" message="info alert" closable/>
      <Alert type="error" message="error alert" closable/>
      <Alert type="warning" message="warning alert" closable/>
    </>
  ))