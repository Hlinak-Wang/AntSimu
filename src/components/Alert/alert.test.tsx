import React from 'react';
import Alert, { AlertType } from './alert';
import { render, fireEvent } from '@testing-library/react';

describe('test set for alert', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('test: render in the correct formate with out description', () => {
    let wrapp = render(<Alert message="message-success" description="description-success" type={AlertType.success}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-info" description="description-info" type={AlertType.info}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-warning" description="description-warning" type={AlertType.warning}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-error" description="description-error" type={AlertType.error}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();
  })

  it('test: render in the correct formate with description', () => {
    let wrapp = render(<Alert message="message-success" type={AlertType.success}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-info" description="description-info" type={AlertType.info}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-warning" description="description-warning" type={AlertType.warning}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();

    wrapp = render(<Alert message="message-error" description="description-error" type={AlertType.error}/>);
    expect(wrapp).toMatchSnapshot();
    wrapp.unmount();
  })

  it('test for closable', () => {
    let closeProps = {
      onClose: jest.fn(),
      afterClose: jest.fn(),
      closable: true
    };
    let wrapp = render(<Alert message="closable" type={AlertType.success} {...closeProps}/>);
    expect(wrapp).toMatchSnapshot();
    let closeButton = wrapp.getByText('X');
    fireEvent.click(closeButton);
    expect(closeProps.onClose).toHaveBeenCalled();
    jest.runAllTimers();
    expect(closeProps.afterClose).toHaveBeenCalled();
    expect(closeButton).not.toBeInTheDocument();
  })
})