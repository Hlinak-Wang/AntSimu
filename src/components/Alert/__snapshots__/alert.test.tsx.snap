// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test set for alert test for closable 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert alert-success alert-closable"
      >
        <span
          class="alert-message"
        >
          closable
        </span>
        <span
          class="alert-description"
        />
        <button
          class="alert-close-button"
        >
          X
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-success alert-closable"
    >
      <span
        class="alert-message"
      >
        closable
      </span>
      <span
        class="alert-description"
      />
      <button
        class="alert-close-button"
      >
        X
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with description 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert alert-success"
      >
        <span
          class="alert-message"
        >
          message-success
        </span>
        <span
          class="alert-description"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-success"
    >
      <span
        class="alert-message"
      >
        message-success
      </span>
      <span
        class="alert-description"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with description 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="alert alert-info"
      >
        <span
          class="alert-message"
        >
          message-info
        </span>
        <span
          class="alert-description"
        >
          description-info
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-info"
    >
      <span
        class="alert-message"
      >
        message-info
      </span>
      <span
        class="alert-description"
      >
        description-info
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with description 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div>
      <div
        class="alert alert-warning"
      >
        <span
          class="alert-message"
        >
          message-warning
        </span>
        <span
          class="alert-description"
        >
          description-warning
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-warning"
    >
      <span
        class="alert-message"
      >
        message-warning
      </span>
      <span
        class="alert-description"
      >
        description-warning
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with description 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div />
    <div>
      <div
        class="alert alert-error"
      >
        <span
          class="alert-message"
        >
          message-error
        </span>
        <span
          class="alert-description"
        >
          description-error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-error"
    >
      <span
        class="alert-message"
      >
        message-error
      </span>
      <span
        class="alert-description"
      >
        description-error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with out description 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert alert-success"
      >
        <span
          class="alert-message"
        >
          message-success
        </span>
        <span
          class="alert-description"
        >
          description-success
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-success"
    >
      <span
        class="alert-message"
      >
        message-success
      </span>
      <span
        class="alert-description"
      >
        description-success
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with out description 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="alert alert-info"
      >
        <span
          class="alert-message"
        >
          message-info
        </span>
        <span
          class="alert-description"
        >
          description-info
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-info"
    >
      <span
        class="alert-message"
      >
        message-info
      </span>
      <span
        class="alert-description"
      >
        description-info
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with out description 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div>
      <div
        class="alert alert-warning"
      >
        <span
          class="alert-message"
        >
          message-warning
        </span>
        <span
          class="alert-description"
        >
          description-warning
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-warning"
    >
      <span
        class="alert-message"
      >
        message-warning
      </span>
      <span
        class="alert-description"
      >
        description-warning
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test set for alert test: render in the correct formate with out description 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div />
    <div>
      <div
        class="alert alert-error"
      >
        <span
          class="alert-message"
        >
          message-error
        </span>
        <span
          class="alert-description"
        >
          description-error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-error"
    >
      <span
        class="alert-message"
      >
        message-error
      </span>
      <span
        class="alert-description"
      >
        description-error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
